install/installer/pkg/components/public-api-server/configmap.go

```diff
@@ -41,13 +42,22 @@ func configmap(ctx *common.RenderContext) ([]runtime.Object, error) {
 		return nil
 	})
 
+	_ = ctx.WithExperimental(func(cfg *experimental.Config) error {
+		publicUrlPrefix = cfg.WebApp.PublicURL
+		if publicUrlPrefix == "" {
+			// we set the default value prefix for api to be "api", and this maintains backward compatibility
+			publicUrlPrefix = "api"
+		}
+		return nil
```

Comment:
What if we just treated this value as a whole value override, rather than a prefix override?

The benefit I see is that it's more predictable what the result will be, and makes our config a bit easier to reason about.

Reply:
In practice, it would mean
```
if the value is set in the config, we use that, otherwise we use `api.DOMAIN`
```

Reply:
that makes sense! updating